### PR TITLE
Remove some outdated mentions of the DataFlagEncoder

### PR DESCRIPTION
**Pull request by easbar**
DataFlagEncoder was removed already for 1.0, but it is still mentioned in a few places

### CODE REVIEW

File: core/src/main/java/com/graphhopper/routing/InstructionsOutgoingEdges.java [hunk 1]
```diff
@@ -139,7 +139,6 @@ public boolean outgoingEdgesAreSlowerByFactor(double factor) {
         for (EdgeIteratorState edge : visibleAlternativeTurns) {
             tmpSpeed = getSpeed(edge);
             if (tmpSpeed < 1) {
-                // This might happen for the DataFlagEncoder, might create unnecessary turn instructions
                 return false;
```
Comment (easbar) [inline core/src/main/java/com/graphhopper/routing/InstructionsOutgoingEdges.java:142]: Do you know if this is still necessary?

Comment (karussell) [reply core/src/main/java/com/graphhopper/routing/InstructionsOutgoingEdges.java:142]: No, unfortunately I do not know. Maybe remove it if no tests fail due to removing?

Comment (easbar) [reply core/src/main/java/com/graphhopper/routing/InstructionsOutgoingEdges.java:142]: @boldtrn maybe?

Comment (easbar) [reply core/src/main/java/com/graphhopper/routing/InstructionsOutgoingEdges.java:142]: The tests are green also after removing it. So let's remove it and put it back if someone can show a case where this is important.

Comment (boldtrn) [reply core/src/main/java/com/graphhopper/routing/InstructionsOutgoingEdges.java]: Not sure if this case can still happen, probably not, so makes sense to remove it :)

Comment (easbar) [reply core/src/main/java/com/graphhopper/routing/InstructionsOutgoingEdges.java]: Ok thanks for taking a look :+1:

**Pull request merged**